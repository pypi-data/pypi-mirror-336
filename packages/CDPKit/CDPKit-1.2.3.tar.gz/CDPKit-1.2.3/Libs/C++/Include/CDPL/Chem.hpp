/* 
 * Chem.hpp 
 *
 * This file is part of the Chemical Data Processing Toolkit
 *
 * Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2 of the License, or (at your option) any later version.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this library; see the file COPYING. If not, write to
 * the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
 * Boston, MA 02111-1307, USA.
 */

/**
 * \file
 * \brief A convenience header including everything that is defined in namespace CDPL::Chem.
 */

#ifndef CDPL_CHEM_HPP
#define CDPL_CHEM_HPP

#include "CDPL/Chem/Entity3D.hpp"
#include "CDPL/Chem/Entity3DContainer.hpp"
#include "CDPL/Chem/AtomContainer.hpp"
#include "CDPL/Chem/BondContainer.hpp"
#include "CDPL/Chem/MolecularGraph.hpp"
#include "CDPL/Chem/Molecule.hpp"
#include "CDPL/Chem/Atom.hpp"
#include "CDPL/Chem/Bond.hpp"
#include "CDPL/Chem/Reaction.hpp"
#include "CDPL/Chem/BasicReaction.hpp"
#include "CDPL/Chem/BasicMolecule.hpp"
#include "CDPL/Chem/BasicAtom.hpp"
#include "CDPL/Chem/BasicBond.hpp"
#include "CDPL/Chem/Fragment.hpp"
#include "CDPL/Chem/ElectronSystem.hpp"
#include "CDPL/Chem/AtomDictionary.hpp"
#include "CDPL/Chem/FragmentList.hpp"
#include "CDPL/Chem/ElectronSystemList.hpp"
#include "CDPL/Chem/Entity3DMapping.hpp"
#include "CDPL/Chem/AtomMapping.hpp"
#include "CDPL/Chem/BondMapping.hpp"
#include "CDPL/Chem/AtomBondMapping.hpp"
#include "CDPL/Chem/StringDataBlock.hpp"
#include "CDPL/Chem/MatchConstraintList.hpp"
#include "CDPL/Chem/StereoDescriptor.hpp"
#include "CDPL/Chem/MultiConfMoleculeInputProcessor.hpp"
#include "CDPL/Chem/DefaultMultiConfMoleculeInputProcessor.hpp"
#include "CDPL/Chem/Atom3DCoordinatesFunction.hpp"
#include "CDPL/Chem/AtomPriorityFunction.hpp"
#include "CDPL/Chem/AtomPredicate.hpp"
#include "CDPL/Chem/BondPredicate.hpp"
#include "CDPL/Chem/AtomCompareFunction.hpp"
#include "CDPL/Chem/BondCompareFunction.hpp"
#include "CDPL/Chem/MatchExpression.hpp"
#include "CDPL/Chem/MatchExpressionList.hpp"
#include "CDPL/Chem/ANDMatchExpressionList.hpp"
#include "CDPL/Chem/ORMatchExpressionList.hpp"
#include "CDPL/Chem/NOTMatchExpression.hpp"
#include "CDPL/Chem/PropertyMatchExpression.hpp"
#include "CDPL/Chem/AtomTypeMatchExpression.hpp"
#include "CDPL/Chem/AtomEnvironmentMatchExpression.hpp"
#include "CDPL/Chem/AtomConfigurationMatchExpression.hpp"
#include "CDPL/Chem/AtomSSSRRingSizeMatchExpression.hpp"
#include "CDPL/Chem/BondConfigurationMatchExpression.hpp"
#include "CDPL/Chem/BondDirectionMatchExpression.hpp"
#include "CDPL/Chem/BondSubstituentDirectionMatchExpression.hpp"
#include "CDPL/Chem/BondReactionCenterStatusMatchExpression.hpp"
#include "CDPL/Chem/MolecularGraphComponentGroupingMatchExpression.hpp"
#include "CDPL/Chem/ReactionAtomMappingMatchExpression.hpp"
#include "CDPL/Chem/ReactionComponentGroupingMatchExpression.hpp"
#include "CDPL/Chem/Entity3DProperty.hpp"
#include "CDPL/Chem/AtomProperty.hpp"
#include "CDPL/Chem/BondProperty.hpp"
#include "CDPL/Chem/MolecularGraphProperty.hpp"
#include "CDPL/Chem/ReactionProperty.hpp"
#include "CDPL/Chem/AtomPropertyDefault.hpp"
#include "CDPL/Chem/BondPropertyDefault.hpp"
#include "CDPL/Chem/MolecularGraphPropertyDefault.hpp"
#include "CDPL/Chem/ReactionPropertyDefault.hpp"
#include "CDPL/Chem/AtomPropertyFlag.hpp"
#include "CDPL/Chem/BondPropertyFlag.hpp"
#include "CDPL/Chem/ControlParameter.hpp"
#include "CDPL/Chem/ControlParameterDefault.hpp"
#include "CDPL/Chem/ControlParameterFunctions.hpp"
#include "CDPL/Chem/AtomConfiguration.hpp"
#include "CDPL/Chem/BondConfiguration.hpp"
#include "CDPL/Chem/CIPDescriptor.hpp"
#include "CDPL/Chem/MDLParity.hpp"
#include "CDPL/Chem/BondStereoFlag.hpp"
#include "CDPL/Chem/AtomType.hpp"
#include "CDPL/Chem/SybylAtomType.hpp"
#include "CDPL/Chem/MOL2MoleculeType.hpp"
#include "CDPL/Chem/MOL2ChargeType.hpp"
#include "CDPL/Chem/HybridizationState.hpp"
#include "CDPL/Chem/ReactionCenterStatus.hpp"
#include "CDPL/Chem/ReactionRole.hpp"
#include "CDPL/Chem/BondDirection.hpp"
#include "CDPL/Chem/RadicalType.hpp"
#include "CDPL/Chem/INCHIReturnCode.hpp"
#include "CDPL/Chem/MDLDataFormatVersion.hpp"
#include "CDPL/Chem/DataFormat.hpp"
#include "CDPL/Chem/TautomerizationType.hpp"
#include "CDPL/Chem/AtomMatchConstraint.hpp"
#include "CDPL/Chem/BondMatchConstraint.hpp"
#include "CDPL/Chem/MolecularGraphMatchConstraint.hpp"
#include "CDPL/Chem/ReactionMatchConstraint.hpp"
#include "CDPL/Chem/INCHIMoleculeInputHandler.hpp"
#include "CDPL/Chem/INCHIMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/JMEMoleculeInputHandler.hpp"
#include "CDPL/Chem/JMEReactionInputHandler.hpp"
#include "CDPL/Chem/JMEMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/JMEReactionOutputHandler.hpp"
#include "CDPL/Chem/MOLMoleculeInputHandler.hpp"
#include "CDPL/Chem/MOLMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/SDFMoleculeInputHandler.hpp"
#include "CDPL/Chem/SDFMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/RXNReactionInputHandler.hpp"
#include "CDPL/Chem/RXNReactionOutputHandler.hpp"
#include "CDPL/Chem/RDFReactionInputHandler.hpp"
#include "CDPL/Chem/RDFReactionOutputHandler.hpp"
#include "CDPL/Chem/SMILESMoleculeInputHandler.hpp"
#include "CDPL/Chem/SMILESReactionInputHandler.hpp"
#include "CDPL/Chem/SMILESMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/SMILESReactionOutputHandler.hpp"
#include "CDPL/Chem/SMARTSMoleculeInputHandler.hpp"
#include "CDPL/Chem/SMARTSReactionInputHandler.hpp"
#include "CDPL/Chem/SMARTSMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/SMARTSReactionOutputHandler.hpp"
#include "CDPL/Chem/CDFMoleculeInputHandler.hpp"
#include "CDPL/Chem/CDFMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/CDFReactionInputHandler.hpp"
#include "CDPL/Chem/CDFReactionOutputHandler.hpp"
#include "CDPL/Chem/MOL2MoleculeInputHandler.hpp"
#include "CDPL/Chem/MOL2MolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/XYZMoleculeInputHandler.hpp"
#include "CDPL/Chem/XYZMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/CMLMoleculeInputHandler.hpp"
#include "CDPL/Chem/CMLMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/SDFGZMoleculeInputHandler.hpp"
#include "CDPL/Chem/SDFGZMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/SDFBZ2MoleculeInputHandler.hpp"
#include "CDPL/Chem/SDFBZ2MolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/SMILESGZMoleculeInputHandler.hpp"
#include "CDPL/Chem/SMILESGZReactionInputHandler.hpp"
#include "CDPL/Chem/SMILESGZMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/SMILESGZReactionOutputHandler.hpp"
#include "CDPL/Chem/SMILESBZ2MoleculeInputHandler.hpp"
#include "CDPL/Chem/SMILESBZ2ReactionInputHandler.hpp"
#include "CDPL/Chem/SMILESBZ2MolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/SMILESBZ2ReactionOutputHandler.hpp"
#include "CDPL/Chem/RDFGZReactionInputHandler.hpp"
#include "CDPL/Chem/RDFGZReactionOutputHandler.hpp"
#include "CDPL/Chem/RDFBZ2ReactionInputHandler.hpp"
#include "CDPL/Chem/RDFBZ2ReactionOutputHandler.hpp"
#include "CDPL/Chem/CDFGZMoleculeInputHandler.hpp"
#include "CDPL/Chem/CDFGZMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/CDFBZ2MoleculeInputHandler.hpp"
#include "CDPL/Chem/CDFBZ2MolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/CDFGZReactionInputHandler.hpp"
#include "CDPL/Chem/CDFGZReactionOutputHandler.hpp"
#include "CDPL/Chem/CDFBZ2ReactionInputHandler.hpp"
#include "CDPL/Chem/CDFBZ2ReactionOutputHandler.hpp"
#include "CDPL/Chem/MOL2GZMoleculeInputHandler.hpp"
#include "CDPL/Chem/MOL2GZMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/MOL2BZ2MoleculeInputHandler.hpp"
#include "CDPL/Chem/MOL2BZ2MolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/XYZGZMoleculeInputHandler.hpp"
#include "CDPL/Chem/XYZGZMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/XYZBZ2MoleculeInputHandler.hpp"
#include "CDPL/Chem/XYZBZ2MolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/CMLGZMoleculeInputHandler.hpp"
#include "CDPL/Chem/CMLGZMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/CMLBZ2MoleculeInputHandler.hpp"
#include "CDPL/Chem/CMLBZ2MolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/SDFGZMoleculeReader.hpp"
#include "CDPL/Chem/SDFGZMolecularGraphWriter.hpp"
#include "CDPL/Chem/SDFBZ2MoleculeReader.hpp"
#include "CDPL/Chem/SDFBZ2MolecularGraphWriter.hpp"
#include "CDPL/Chem/RDFGZReactionReader.hpp"
#include "CDPL/Chem/RDFGZReactionWriter.hpp"
#include "CDPL/Chem/RDFBZ2ReactionReader.hpp"
#include "CDPL/Chem/RDFBZ2ReactionWriter.hpp"
#include "CDPL/Chem/SMILESGZMoleculeReader.hpp"
#include "CDPL/Chem/SMILESGZMolecularGraphWriter.hpp"
#include "CDPL/Chem/SMILESBZ2MoleculeReader.hpp"
#include "CDPL/Chem/SMILESBZ2MolecularGraphWriter.hpp"
#include "CDPL/Chem/SMILESGZReactionReader.hpp"
#include "CDPL/Chem/SMILESGZReactionWriter.hpp"
#include "CDPL/Chem/SMILESBZ2ReactionReader.hpp"
#include "CDPL/Chem/SMILESBZ2ReactionWriter.hpp"
#include "CDPL/Chem/CDFGZMoleculeReader.hpp"
#include "CDPL/Chem/CDFGZMolecularGraphWriter.hpp"
#include "CDPL/Chem/CDFBZ2MoleculeReader.hpp"
#include "CDPL/Chem/CDFBZ2MolecularGraphWriter.hpp"
#include "CDPL/Chem/CDFGZReactionReader.hpp"
#include "CDPL/Chem/CDFGZReactionWriter.hpp"
#include "CDPL/Chem/CDFBZ2ReactionReader.hpp"
#include "CDPL/Chem/CDFBZ2ReactionWriter.hpp"
#include "CDPL/Chem/MOL2GZMoleculeReader.hpp"
#include "CDPL/Chem/MOL2GZMolecularGraphWriter.hpp"
#include "CDPL/Chem/MOL2BZ2MoleculeReader.hpp"
#include "CDPL/Chem/MOL2BZ2MolecularGraphWriter.hpp"
#include "CDPL/Chem/XYZGZMoleculeReader.hpp"
#include "CDPL/Chem/XYZGZMolecularGraphWriter.hpp"
#include "CDPL/Chem/XYZBZ2MoleculeReader.hpp"
#include "CDPL/Chem/XYZBZ2MolecularGraphWriter.hpp"
#include "CDPL/Chem/CMLGZMoleculeReader.hpp"
#include "CDPL/Chem/CMLGZMolecularGraphWriter.hpp"
#include "CDPL/Chem/CMLBZ2MoleculeReader.hpp"
#include "CDPL/Chem/CMLBZ2MolecularGraphWriter.hpp"
#include "CDPL/Chem/JMEMoleculeReader.hpp"
#include "CDPL/Chem/JMEReactionReader.hpp"
#include "CDPL/Chem/JMEMolecularGraphWriter.hpp"
#include "CDPL/Chem/JMEReactionWriter.hpp"
#include "CDPL/Chem/SMILESMoleculeReader.hpp"
#include "CDPL/Chem/SMILESReactionReader.hpp"
#include "CDPL/Chem/SMILESMolecularGraphWriter.hpp"
#include "CDPL/Chem/SMILESReactionWriter.hpp"
#include "CDPL/Chem/SMARTSMoleculeReader.hpp"
#include "CDPL/Chem/SMARTSReactionReader.hpp"
#include "CDPL/Chem/SMARTSMolecularGraphWriter.hpp"
#include "CDPL/Chem/SMARTSReactionWriter.hpp"
#include "CDPL/Chem/MOLMoleculeReader.hpp"
#include "CDPL/Chem/MOLMolecularGraphWriter.hpp"
#include "CDPL/Chem/SDFMoleculeReader.hpp"
#include "CDPL/Chem/SDFMolecularGraphWriter.hpp"
#include "CDPL/Chem/RXNReactionReader.hpp"
#include "CDPL/Chem/RXNReactionWriter.hpp"
#include "CDPL/Chem/RDFReactionReader.hpp"
#include "CDPL/Chem/RDFReactionWriter.hpp"
#include "CDPL/Chem/INCHIMoleculeReader.hpp"
#include "CDPL/Chem/INCHIMolecularGraphWriter.hpp"
#include "CDPL/Chem/CDFMoleculeReader.hpp"
#include "CDPL/Chem/CDFMolecularGraphWriter.hpp"
#include "CDPL/Chem/CDFReactionReader.hpp"
#include "CDPL/Chem/CDFReactionWriter.hpp"
#include "CDPL/Chem/MOL2MoleculeReader.hpp"
#include "CDPL/Chem/MOL2MolecularGraphWriter.hpp"
#include "CDPL/Chem/XYZMoleculeReader.hpp"
#include "CDPL/Chem/XYZMolecularGraphWriter.hpp"
#include "CDPL/Chem/CMLMoleculeReader.hpp"
#include "CDPL/Chem/CMLMolecularGraphWriter.hpp"
#include "CDPL/Chem/MoleculeReader.hpp"
#include "CDPL/Chem/MolecularGraphWriter.hpp"
#include "CDPL/Chem/ReactionReader.hpp"
#include "CDPL/Chem/ReactionWriter.hpp"
#include "CDPL/Chem/CompleteRingSet.hpp"
#include "CDPL/Chem/AromaticRingSet.hpp"
#include "CDPL/Chem/AromaticSSSRSubset.hpp"
#include "CDPL/Chem/SmallestSetOfSmallestRings.hpp"
#include "CDPL/Chem/ExtendedSSSR.hpp"
#include "CDPL/Chem/AromaticSubstructure.hpp"
#include "CDPL/Chem/CyclicSubstructure.hpp"
#include "CDPL/Chem/ConnectedSubstructureSet.hpp"
#include "CDPL/Chem/ComponentSet.hpp"
#include "CDPL/Chem/PiElectronSystemList.hpp"
#include "CDPL/Chem/SubstructureSearch.hpp"
#include "CDPL/Chem/ReactionSubstructureSearch.hpp"
#include "CDPL/Chem/CommonConnectedSubstructureSearch.hpp"
#include "CDPL/Chem/MaxCommonAtomSubstructureSearch.hpp"
#include "CDPL/Chem/MaxCommonBondSubstructureSearch.hpp"
#include "CDPL/Chem/AutomorphismGroupSearch.hpp"
#include "CDPL/Chem/MultiSubstructureSearch.hpp"
#include "CDPL/Chem/Reactor.hpp"
#include "CDPL/Chem/MorganNumberingCalculator.hpp"
#include "CDPL/Chem/CIPPriorityCalculator.hpp"
#include "CDPL/Chem/CIPConfigurationLabeler.hpp"
#include "CDPL/Chem/CanonicalNumberingCalculator.hpp"
#include "CDPL/Chem/HashCodeCalculator.hpp"
#include "CDPL/Chem/Atom2DCoordinatesCalculator.hpp"
#include "CDPL/Chem/Hydrogen3DCoordinatesCalculator.hpp"
#include "CDPL/Chem/BondStereoFlagCalculator.hpp"
#include "CDPL/Chem/BondOrderCalculator.hpp"
#include "CDPL/Chem/KekuleStructureCalculator.hpp"
#include "CDPL/Chem/SymmetryClassCalculator.hpp"
#include "CDPL/Chem/SurfaceAtomExtractor.hpp"
#include "CDPL/Chem/PatternAtomTyper.hpp"
#include "CDPL/Chem/SubstructureHistogramCalculator.hpp"
#include "CDPL/Chem/TautomerScore.hpp"
#include "CDPL/Chem/TautomerGenerator.hpp"
#include "CDPL/Chem/DefaultTautomerGenerator.hpp"
#include "CDPL/Chem/TautomerizationRule.hpp"
#include "CDPL/Chem/PatternBasedTautomerizationRule.hpp"
#include "CDPL/Chem/KetoEnolTautomerization.hpp"
#include "CDPL/Chem/ImineEnamineTautomerization.hpp"
#include "CDPL/Chem/NitrosoOximeTautomerization.hpp"
#include "CDPL/Chem/AmideImidicAcidTautomerization.hpp"
#include "CDPL/Chem/LactamLactimTautomerization.hpp"
#include "CDPL/Chem/KeteneYnolTautomerization.hpp"
#include "CDPL/Chem/NitroAciTautomerization.hpp"
#include "CDPL/Chem/PhosphinicAcidTautomerization.hpp"
#include "CDPL/Chem/SulfenicAcidTautomerization.hpp"
#include "CDPL/Chem/GenericHydrogen13ShiftTautomerization.hpp"
#include "CDPL/Chem/GenericHydrogen15ShiftTautomerization.hpp"
#include "CDPL/Chem/ResonanceStructureGenerator.hpp"
#include "CDPL/Chem/StereoisomerGenerator.hpp"
#include "CDPL/Chem/ChEMBLStandardizer.hpp"
#include "CDPL/Chem/ProtonationStateStandardizer.hpp"
#include "CDPL/Chem/FragmentGenerator.hpp"
#include "CDPL/Chem/RECAPRuleID.hpp"
#include "CDPL/Chem/RECAPAtomLabel.hpp"
#include "CDPL/Chem/RECAPFragmentGenerator.hpp"
#include "CDPL/Chem/BRICSRuleID.hpp"
#include "CDPL/Chem/BRICSAtomLabel.hpp"
#include "CDPL/Chem/BRICSFragmentGenerator.hpp"
#include "CDPL/Chem/BemisMurckoAnalyzer.hpp"
#include "CDPL/Chem/TopologicalEntityAlignment.hpp"
#include "CDPL/Chem/SpatialEntityAlignment.hpp"
#include "CDPL/Chem/Atom3DCoordinatesFunctor.hpp"
#include "CDPL/Chem/AtomConformer3DCoordinatesFunctor.hpp"
#include "CDPL/Chem/AtomArray3DCoordinatesFunctor.hpp"
#include "CDPL/Chem/UtilityFunctions.hpp"
#include "CDPL/Chem/AtomContainerFunctions.hpp"
#include "CDPL/Chem/Entity3DContainerFunctions.hpp"
#include "CDPL/Chem/AtomFunctions.hpp"
#include "CDPL/Chem/BondFunctions.hpp"
#include "CDPL/Chem/MolecularGraphFunctions.hpp"
#include "CDPL/Chem/MoleculeFunctions.hpp"
#include "CDPL/Chem/FragmentFunctions.hpp"
#include "CDPL/Chem/ReactionFunctions.hpp"

#endif // CDPL_CHEM_HPP
