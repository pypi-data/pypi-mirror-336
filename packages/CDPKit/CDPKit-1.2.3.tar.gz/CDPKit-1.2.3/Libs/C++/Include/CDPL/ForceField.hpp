/* 
 * ForceField.hpp 
 *
 * This file is part of the Chemical Data Processing Toolkit
 *
 * Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2 of the License, or (at your option) any later version.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this library; see the file COPYING. If not, write to
 * the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
 * Boston, MA 02111-1307, USA.
 */

/**
 * \file
 * \brief A convenience header including everything that is defined in namespace CDPL::ForceField.
 */

#ifndef CDPL_FORCEFIELD_HPP
#define CDPL_FORCEFIELD_HPP

#include "CDPL/ForceField/Exceptions.hpp"
#include "CDPL/ForceField/AtomProperty.hpp"
#include "CDPL/ForceField/BondProperty.hpp"
#include "CDPL/ForceField/MolecularGraphProperty.hpp"
#include "CDPL/ForceField/AtomFunctions.hpp"
#include "CDPL/ForceField/BondFunctions.hpp"
#include "CDPL/ForceField/MolecularGraphFunctions.hpp"
#include "CDPL/ForceField/UtilityFunctions.hpp"
#include "CDPL/ForceField/InteractionFilterFunctions.hpp"
#include "CDPL/ForceField/TopologicalAtomDistanceFunction.hpp"
#include "CDPL/ForceField/InteractionType.hpp"
#include "CDPL/ForceField/GradientVectorTraits.hpp"
#include "CDPL/ForceField/UFFAtomType.hpp"
#include "CDPL/ForceField/UFFAtomTypePropertyTable.hpp"
#include "CDPL/ForceField/MMFF94ParameterSet.hpp"
#include "CDPL/ForceField/MMFF94PropertyFunctions.hpp"
#include "CDPL/ForceField/MMFF94AromaticSSSRSubset.hpp"
#include "CDPL/ForceField/MMFF94AtomTyper.hpp"
#include "CDPL/ForceField/MMFF94BondTyper.hpp"
#include "CDPL/ForceField/MMFF94ChargeCalculator.hpp"
#include "CDPL/ForceField/MMFF94EnergyFunctions.hpp"
#include "CDPL/ForceField/MMFF94GradientFunctions.hpp"
#include "CDPL/ForceField/MMFF94EnergyCalculator.hpp"
#include "CDPL/ForceField/MMFF94GradientCalculator.hpp"
#include "CDPL/ForceField/MMFF94BondStretchingInteraction.hpp"
#include "CDPL/ForceField/MMFF94AngleBendingInteraction.hpp"
#include "CDPL/ForceField/MMFF94StretchBendInteraction.hpp"
#include "CDPL/ForceField/MMFF94OutOfPlaneBendingInteraction.hpp"
#include "CDPL/ForceField/MMFF94TorsionInteraction.hpp"
#include "CDPL/ForceField/MMFF94VanDerWaalsInteraction.hpp"
#include "CDPL/ForceField/MMFF94ElectrostaticInteraction.hpp"
#include "CDPL/ForceField/MMFF94BondStretchingInteractionList.hpp"
#include "CDPL/ForceField/MMFF94AngleBendingInteractionList.hpp"
#include "CDPL/ForceField/MMFF94StretchBendInteractionList.hpp"
#include "CDPL/ForceField/MMFF94OutOfPlaneBendingInteractionList.hpp"
#include "CDPL/ForceField/MMFF94TorsionInteractionList.hpp"
#include "CDPL/ForceField/MMFF94VanDerWaalsInteractionList.hpp"
#include "CDPL/ForceField/MMFF94ElectrostaticInteractionList.hpp"
#include "CDPL/ForceField/MMFF94InteractionData.hpp"
#include "CDPL/ForceField/MMFF94BondStretchingInteractionParameterizer.hpp"
#include "CDPL/ForceField/MMFF94AngleBendingInteractionParameterizer.hpp"
#include "CDPL/ForceField/MMFF94StretchBendInteractionParameterizer.hpp"
#include "CDPL/ForceField/MMFF94OutOfPlaneBendingInteractionParameterizer.hpp"
#include "CDPL/ForceField/MMFF94TorsionInteractionParameterizer.hpp"
#include "CDPL/ForceField/MMFF94VanDerWaalsInteractionParameterizer.hpp"
#include "CDPL/ForceField/MMFF94ElectrostaticInteractionParameterizer.hpp"
#include "CDPL/ForceField/MMFF94InteractionParameterizer.hpp"
#include "CDPL/ForceField/MMFF94SymbolicAtomTypePatternTable.hpp"
#include "CDPL/ForceField/MMFF94HeavyToHydrogenAtomTypeMap.hpp"
#include "CDPL/ForceField/MMFF94SymbolicToNumericAtomTypeMap.hpp"
#include "CDPL/ForceField/MMFF94AromaticAtomTypeDefinitionTable.hpp"
#include "CDPL/ForceField/MMFF94AtomTypePropertyTable.hpp"
#include "CDPL/ForceField/MMFF94FormalAtomChargeDefinitionTable.hpp"
#include "CDPL/ForceField/MMFF94BondChargeIncrementTable.hpp"
#include "CDPL/ForceField/MMFF94PartialBondChargeIncrementTable.hpp"
#include "CDPL/ForceField/MMFF94PrimaryToParameterAtomTypeMap.hpp"
#include "CDPL/ForceField/MMFF94AngleBendingParameterTable.hpp"
#include "CDPL/ForceField/MMFF94BondStretchingParameterTable.hpp"
#include "CDPL/ForceField/MMFF94BondStretchingRuleParameterTable.hpp"
#include "CDPL/ForceField/MMFF94StretchBendParameterTable.hpp"
#include "CDPL/ForceField/MMFF94DefaultStretchBendParameterTable.hpp"
#include "CDPL/ForceField/MMFF94OutOfPlaneBendingParameterTable.hpp"
#include "CDPL/ForceField/MMFF94TorsionParameterTable.hpp"
#include "CDPL/ForceField/MMFF94VanDerWaalsParameterTable.hpp"
#include "CDPL/ForceField/ElasticPotential.hpp"
#include "CDPL/ForceField/ElasticPotentialList.hpp"
#include "CDPL/ForceField/ElasticPotentialFunctions.hpp"

#endif // CDPL_FORCEFIELD_HPP
