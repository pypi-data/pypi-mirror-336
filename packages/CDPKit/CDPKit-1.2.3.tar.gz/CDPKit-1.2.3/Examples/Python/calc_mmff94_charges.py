#!/bin/env python

##
# calc_mmff94_charges.py 
#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
#
# Permission to use, copy, modify, and/or distribute this software for any purpose with
# or without fee is hereby granted.
# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH REGARD
# TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS. IN
# NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL
# DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER
# IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
# CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
##


import sys

import CDPL.Chem as Chem
import CDPL.ForceField as ForceField


# calculates and outputs the MMFF94 charges of the atoms of the provided molecular graph
def calcAndOutputCharges(mol: Chem.Molecule) -> None:
    Chem.calcImplicitHydrogenCounts(mol, False)  # calculate implicit hydrogen counts and set corresponding property for all atoms
    Chem.makeHydrogenComplete(mol)               # make all implicit hydrogens explicit
    Chem.perceiveHybridizationStates(mol, False) # perceive atom hybridization states and set corresponding property for all atoms
    Chem.perceiveSSSR(mol, False)                # perceive smallest set of smallest rings and store as Chem.MolecularGraph property
    Chem.setRingFlags(mol, False)                # perceive cycles and set corresponding atom and bond properties
    Chem.setAromaticityFlags(mol, False)         # perceive aromaticity and set corresponding atom and bond properties
  
    ForceField.perceiveMMFF94AromaticRings(mol, False)        # perceive aromatic rings according to the MMFF94 aroamticity model and store data as Chem.MolecularGraph property
    ForceField.assignMMFF94AtomTypes(mol, False, False)       # perceive MMFF94 atom types (tolerant mode) set corresponding property for all atoms
    ForceField.assignMMFF94BondTypeIndices(mol, False, False) # perceive MMFF94 bond types (tolerant mode) set corresponding property for all bonds
    ForceField.calcMMFF94AtomCharges(mol, False, False)       # calculate MMFF94 atom charges (tolerant mode) set corresponding property for all atoms

    print('- MMFF94 partial charges')
    
    for atom in mol.atoms:
        print('Atom #%s: %s' % (str(atom.getIndex()), str(ForceField.getMMFF94Charge(atom))))

def main() -> None:
    if len(sys.argv) < 2:
        sys.exit('Usage: %s <input mol. file>' % sys.argv[0])

    # create reader for input molecules (format specified by file extension)
    reader = Chem.MoleculeReader(sys.argv[1]) 
 
    # create an instance of the default implementation of the Chem.Molecule interface
    mol = Chem.BasicMolecule()
    
    # read and process molecules one after the other until the end of input has been reached
    try:
        while reader.read(mol):
            try:
                calcAndOutputCharges(mol)
            except Exception as e:
                sys.exit('Error: processing of molecule failed: ' + str(e))
                
    except Exception as e: # handle exception raised in case of severe read errors
        sys.exit('Error: reading molecule failed: ' + str(e))

    sys.exit(0)
        
if __name__ == '__main__':
    main()
