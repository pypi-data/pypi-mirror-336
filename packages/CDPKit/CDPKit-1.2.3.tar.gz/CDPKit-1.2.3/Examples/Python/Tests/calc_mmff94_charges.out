- MMFF94 partial charges
Atom #0: -0.19
Atom #1: 0.19
Atom #2: -0.15
Atom #3: -0.15
Atom #4: -0.1435
Atom #5: 0.567
Atom #6: -0.56
Atom #7: 0.4235
Atom #8: -0.1435
Atom #9: -0.1435
Atom #10: -0.15
Atom #11: -0.15
Atom #12: 0.073
Atom #13: -0.15
Atom #14: 0.15
Atom #15: 0.48410000000000003
Atom #16: -0.5571
Atom #17: 0.15
Atom #18: 0.15
Atom #19: 0.0
Atom #20: 0.0
Atom #21: 0.0
Atom #22: 0.0
Atom #23: 0.27
Atom #24: -0.81
Atom #25: 0.27
Atom #26: 0.0
Atom #27: 0.0
Atom #28: 0.0
Atom #29: 0.27
Atom #30: 0.0
Atom #31: 0.0
Atom #32: 0.0
Atom #33: 0.0
Atom #34: 0.0
Atom #35: 0.0
Atom #36: 0.0
Atom #37: 0.0
Atom #38: 0.0
Atom #39: -0.15
Atom #40: -0.15
Atom #41: 0.15
Atom #42: 0.15
Atom #43: 0.15
Atom #44: 0.15
